File: backend/src/main/java/org/nextpertis/saasfeedbackcollector/service/FeedbackService.java
package org.nextpertis.saasfeedbackcollector.service;

import org.nextpertis.saasfeedbackcollector.dto.FeedbackRequest;
import org.nextpertis.saasfeedbackcollector.model.Feedback;

import java.util.List;

public interface FeedbackService {
    Feedback createFeedback(FeedbackRequest feedbackRequest);

    Feedback getFeedbackById(String id);

    List<Feedback> getFeedbacksByCategory(String category);
} 

File: frontend/src/app/services/feedback.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { Feedback, FeedbackRequest } from '../models/feedback.interface';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class FeedbackService {
  private apiUrl = `${environment.apiUrl}/feedback`;

  constructor(private http: HttpClient) {}

  submitFeedback(feedback: FeedbackRequest): Observable<Feedback> {
    return this.http.post<Feedback>(this.apiUrl, feedback);
  }

  getFeedbacksByCategory(category: string): Observable<Feedback[]> {
    return this.http.get<Feedback[]>(`${this.apiUrl}?category=${category}`);
  }

  getFeedbackById(id: string): Observable<Feedback> {
    return this.http.get<Feedback>(`${this.apiUrl}/${id}`);
  }
}
